subsys/mgmt/settings/src/settings_mgmt.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <logging/log.h>
+#include <mgmt/mgmt.h>
+#include <mgmt/settings_mgmt.h>
+#include <cborattr/cborattr.h>
+#include <stats/stats.h>
+
+LOG_MODULE_REGISTER(mgmt_settings, CONFIG_MGMT_SETTINGS_LOG_LEVEL);
+
+#define SETTINGS_MGMT_ID_ERASE 6
+
+static int settings_erase(struct mgmt_ctxt *ctxt)
+{
+	int rc = -EFAULT;
+	CborError cbor_err = 0;
+
+#ifdef PM_SETTINGS_STORAGE_ID
```

Comment:
Is there any reason for this mgmt to be included in a project if there is no storage partition? I think this check should be done in the build system so that its not needed in this source file as we should assume that this partition is defined at this point.

You should use the `fash_map.h` defines to be compatible with DT as well https://github.com/zephyrproject-rtos/zephyr/blob/main/include/storage/flash_map.h#L265

Reply:
Sure, the check can be done in the build system. This was posted a bit too early as a PR it was supposed to be a draft :)